coa/pkg/apis/v1alpha2/providers/stringlock/stringlock.go

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ * SPDX-License-Identifier: MIT
+ */
+
+package stringlock
+
+import (
+	providers "github.com/eclipse-symphony/symphony/coa/pkg/apis/v1alpha2/providers"
+	//"encoding/json"
+)
+
+type UnLock func()
+
+type IStringLockProvider interface {
+	Init(config providers.IProviderConfig) error
+	Lock(string) UnLock
```

Comment:
This interface looks strange to me. Usually Lock function should return an error indicating whether lock is successful or not. And another Unlock function in the interface to unlock a key.

Reply:
added "trylock" and "unlock" interface

type IKeyLockProvider interface {
	Init(config providers.IProviderConfig) error
	Lock(string)
	UnLock(string)
	TryLock(string) bool
	TryLockWithTimeout(string, time.Duration) bool
}